src/core.js

```diff
@@ -282,12 +283,12 @@ function done() {
 		runtime: runtime
 	} );
 
-	// Clear own sessionStorage items if all tests passed
-	if ( config.reorder && defined.sessionStorage && config.stats.bad === 0 ) {
-		for ( i = 0; i < sessionStorage.length; i++ ) {
-			key = sessionStorage.key( i++ );
+	// Clear own storage items if all tests passed
+	if ( config.reorder && storage && config.stats.bad === 0 ) {
+		for ( i = 0; i < storage.length; i++ ) {
+			key = storage.key( i++ );
```

Comment:
Are we going to let this slide again, or finally fix it? The right logic would look like
```js
for ( i = storage.length - 1; i >= 0; i-- ) {
	key = storage.key( i );
	if ( … ) {
		storage.removeItem( key );
	}
}
```